packages/waku/src/router/define-router.ts

```diff
@@ -269,3 +269,33 @@ export function unstable_rerenderRoute(
   const rscPath = getRscPath(pathname);
   rerender(rscPath, { query, skip });
 }
+
+// -----------------------------------------------------
+// new_defineRouter
+// Eventually replaces unstable_defineRouter
+// -----------------------------------------------------
+
+export function new_defineRouter(
+  getPathConfig: () => Promise<
+    Iterable<{
+      pattern: string; // TODO let's revisit this later
+      path: PathSpec;
+      components: Record<
+        string, // componentId such as "**/layout" or "**/page" in the case of createPages
+        {
+          isStatic?: boolean;
+        }
+      >;
+      noSsr?: boolean;
+    }>
+  >,
+  renderRoute: (
```

Comment:
This is a small dx idea, but I think it might be helpful to have the params of `defineRouter` to be named in an object param. Something like:

```ts
export default defineEntries({
  renderEntries: async (rscPath) => {
    return {
      App: <App name={rscPath || 'Waku'} />,
    };
  },
  getBuildConfig: async () => [{ pathname: '/', entries: [{ rscPath: '' }] }],
  getSSRConfig: () => {
    throw new Error('SSR is should not be used in this test.');
  },
});
```

just to make it clear what each thing is doing from the consumer perspective without needing to rely on comments

Reply:
yeah, we can do that. I'll redesign defineEntries too, later.